plugins/pur/repository-proxy/src/main/java/org/pentaho/di/repository/pur/provider/PurRepositoryProxy.java

```diff
@@ -0,0 +1,717 @@
+import org.pentaho.di.core.exception.KettleSecurityException;
+import org.pentaho.di.core.logging.LogChannelInterface;
+import org.pentaho.di.core.plugins.PluginInterface;
+import org.pentaho.di.core.plugins.PluginRegistry;
+import org.pentaho.di.core.plugins.RepositoryPluginType;
+import org.pentaho.di.job.JobMeta;
+import org.pentaho.di.partition.PartitionSchema;
+import org.pentaho.di.repository.IRepositoryExporter;
+import org.pentaho.di.repository.IRepositoryImporter;
+import org.pentaho.di.repository.IRepositoryService;
+import org.pentaho.di.repository.IUser;
+import org.pentaho.di.repository.ObjectId;
+import org.pentaho.di.repository.Repository;
+import org.pentaho.di.repository.RepositoryDirectoryInterface;
+import org.pentaho.di.repository.RepositoryElementInterface;
+import org.pentaho.di.repository.RepositoryElementMetaInterface;
+import org.pentaho.di.repository.RepositoryMeta;
+import org.pentaho.di.repository.RepositoryObject;
+import org.pentaho.di.repository.RepositoryObjectType;
+import org.pentaho.di.repository.RepositorySecurityManager;
+import org.pentaho.di.repository.RepositorySecurityProvider;
+import org.pentaho.di.shared.SharedObjects;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.metastore.api.IMetaStore;
+import org.pentaho.platform.api.repository2.unified.IUnifiedRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Calendar;
+import java.util.List;
+
+/**
+ * Provides a bridge between OSGI and non OSGI worlds by delegating the calls to PurRepository, to
+ * the repository instance loaded from the plugin class loader.
+ */
+public class PurRepositoryProxy implements Repository {
+
+  private static final Logger logger = LoggerFactory.getLogger( PurRepositoryProxy.class );
+  private static final String PUR_PLUGIN_ID = "PentahoEnterpriseRepository";
+  private ClassLoader purPluginClassLoader;
+  private Repository delegate;
+  private String locationUrl;
+
+  public PurRepositoryProxy( PluginRegistry registry ) throws KettlePluginException {
+    PluginInterface purPlugin = registry.findPluginWithId( RepositoryPluginType.class, PUR_PLUGIN_ID );
+    purPluginClassLoader = registry.getClassLoader( purPlugin );
+  }
+
+  protected Repository getDelegate() {
+    if ( this.delegate != null ) {
+      return this.delegate;
+    }
+    Repository repository = null;
+    try {
+      repository = (Repository) purPluginClassLoader.loadClass( "org.pentaho.di.repository.pur.PurRepository" ).newInstance();
+    } catch ( Exception e ) {
+      logger.error( "Unable to load delegate class for plugin id \"{}\". PUR plugin is most likely not installed or "
+          + "changed id.", PUR_PLUGIN_ID );
+    }
+    return this.delegate = repository;
+  }
+
+  /**
+   * @param locationUrl
+   */
+  public void setLocationUrl( String locationUrl ) {
+//    this.getDelegate();
```

Comment:
If this is just forgotten commented-out old code, it should be removed. If not, please add a comment to explain why this is here.

Reply:
The comment was forgotten there. It's been removed